Associate any organization user to a space.

o-orand opened the pull request:
Add userdGuid to  associatexxxxWithSpace method, if null, use current user.
Fix associatexxxxWithSpace orgName param used to search for space: if null use current org, else use orgName.
Add getOrganizationUsers to be able to retrieve all users of an organization indexed by username.
Change associatexxxxWithSpace method signatures introduced with #271 - getSpaceDevelopers(String spaceName) returns NPE in version 1.1.1

This closes #278 - As CLI does, add any user to a space.

scottfrederick commented on cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudFoundryClient.java:
```diff
@@ -25,26 +25,8 @@
-import org.cloudfoundry.client.lib.domain.ApplicationLog;
-import org.cloudfoundry.client.lib.domain.ApplicationStats;
-import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.*;
```
Please don't re-organize imports to use wildcards.

scottfrederick commented on cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudFoundryOperations.java:
```diff
@@ -16,6 +16,11 @@
 package org.cloudfoundry.client.lib;
 
+import org.cloudfoundry.client.lib.archive.ApplicationArchive;
+import org.cloudfoundry.client.lib.domain.*;
```
Please don't re-organize imports to use wildcards.

scottfrederick commented on cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudFoundryOperations.java:
```diff
@@ -159,29 +141,32 @@
-	 * @param spaceName name of the space
-	 */
-	void associateManagerWithSpace(String orgName, String spaceName);
+    /**
```
These lines appear to be indented with spaces instead of tabs.

scottfrederick commented:
Thanks for the PR @o-orand. 

Adding the third `String userGuid` parameter to the `associateXxxxWithSpace` methods makes this a breaking API change, which we try to avoid with point releases. For backward compatibility, it would be better to leave the existing methods intact and add additional methods that have the third parameter.

o-orand commented:
@scottfrederick, thanks for feedback. I've updated the PR. I fixed formatting issues (IDE default...) and I restored ```associateXxxxWithSpace``` 2 params version to restore backward compatibility.
